_posts/en/newsletters/2018-11-06-newsletter.md

```diff
@@ -0,0 +1,177 @@
+
+- **Discussion about improving Lightning payments:** in advance of an
+  upcoming meeting between LN protocol developers, Rusty Russel started
+  a [discussion][ln bolt11 ss] about two problems he thinks could
+  potentially be solved using scriptless scripts as described in
+  [Newsletter #16][].
+
+    1. An invoice can only be paid a maximum of one time.  It'd be nice
+       for multiple people to be able to pay the same invoice, such as a
+       static donation invoice or a monthly recurrent payment.
+
+    2. The protocol can't provide proof of payment.  You can prove that
+       a particular invoice was paid, and that invoice could commit to
+       the identity of the person who was supposed to pay it, but both
+       the payer and the nodes who help route the payment to the
+       recipient all have the same data about the payment, so any one of
+       them could claim to have sent the payment themselves.
+
+## Optech recommends
+
+If you're looking for more news about Lightning, check out Rene
+Pickhardt's new weekly collection of the best tweets about LN and what
+people are building with it.  Follow [@renepickhardt][] on Twitter to
+get the latest news and check out the previously published issues:
+[1][lwil41], [2][lwil42], and [3][lwil43].
+
+## Notable code changes
+
+*Notable code changes this week in [Bitcoin Core][core commits],
+[LND][lnd commits], [C-lightning][cl commits], and [libsecp256k1][secp
+commits].*
+
+{% include linkers/github-log.md
+  refname="core commits"
+  repo="bitcoin/bitcoin"
+  start="f1e2f2a85962c1664e4e55471061af0eaa798d40"
+  end="742ee213499194f97e59dae4971f1474ae7d57ad"
+%}
+{% include linkers/github-log.md
+  refname="lnd commits"
+  repo="lightningnetwork/lnd"
+  start="e5b84cfadab56037ae3957e704b3e570c9368297"
+  end="6b19df162a161079ab794162b45e8f4c7bb8beec"
+%}
+{% include linkers/github-log.md
+  refname="cl commits"
+  repo="ElementsProject/lightning"
+  start="22b8a88b488faa94a009b2c58415ae825152f709"
+  end="d5bb536ef0c08a813f767b3fb016eb20292de4dd"
+%}
+{% include linkers/github-log.md
+  refname="secp commits"
+  repo="bitcoin-core/secp256k1"
+  start="1086fda4c1975d0cad8d3cad96794a64ec12dca4"
+  end="1086fda4c1975d0cad8d3cad96794a64ec12dca4"
+%}
+
+- [Bitcoin Core #14454][] adds support to the [importmulti][rpc
+  importmulti] RPC for segwit addresses and scripts (P2WPKH, P2WSH, and
+  P2SH-wrapped segwit).  A new `witnessscript` parameter fulfills the
+  same role for segwit as the `redeemscript` parameter for P2SH.  Also a
+  `solvable` parameter is added to the [getaddressinfo][rpc
+  getaddressinfo] RPC to let the user know whether the wallet knows the
+  redeemScript or witnessScript for a P2SH or P2WSH address, i.e.
+  whether it knows how to create an unsigned input for spending payments
+  sent to that address.
+
+- [LND #2027][] adds a configuration option that allows a node to reject
+  new channels being opened with an initial "push" of funds.  This is
```

Comment:
Where did you get this motivation? The initial issue was opened because customers were accidentally pushing payment to bitrefill when opening a channel: (https://github.com/lightningnetwork/lnd/issues/1884)

Reply:
1. I didn't mean to suggest that as a motivation for this change.  I myself didn't see a clearly-stated motivation for the change and invented that example as a plausible use for this feature.  I'd be happy to learn of a different motivation or to remove the suggested benefit from the text.

2. I saw that issue in my research and, double checking it now, I seem to be consistently reading it differently than you.  It sounds like you're saying the problem is "Alice accidentally *paid* Bob when opening a channel to him", but I think the problem is "Alice accidentally *opened a channel* to Bob when paying him".  I think the issue (and the PR fixing it) were about finding a way to keep payments in an existing channel rather than having users open a new channel; I think the documentation in the PR supports this: "`rejectpush`: If true, lnd will not accept channel opening requests with non-zero push amounts. This should prevent accidental pushes to merchant nodes."

    I think this option makes sense for a merchant that only wants users to have an experience of fast payments, as pay-in-a-new-channel payments have to wait for the configured number of confirmations like any onchain transaction.

Reply:
@jamesob and I have discussed this, and I still think my interpretation is correct. Here's what I guess is happening at bitrefill:

- customer goes through the shopping process on bitrefill. Selects lightning payment and is presented with a screen saying "send 100000 satoshis. Here's a lightning invoice: lnbcabc123"
- the customer doesn't have any channels open, so they decide to open a channel with bitrefill with an initial push of 100000 to pay for the order
- that push is not a payment for the invoice. The customer thinks they've paid, but the merchant doesn't see a paid invoice.

A merchant would want all payments over lightning to be associated with an invoice, so they would never want customers opening funds with an initial push.

This is just speculation though. It'd be great if @juscamarena or @mrwhythat would confirm for us!

Reply:
@jnewbery is correct with his interpretation.

I opened the issue to have the option to prevent users or even routing nodes from being able to push an amount when creating a channel as it's almost always done mistakenly.

I and other node operators have seen users mistakenly push funds, for a public service that means having to refund the user, for nodes without a public identity it's free money!

At bitrefill did want to have some sort of push and pay while opening a channel but it's a bit trickier without an invoice, unless we have an out of bound acknowledgment that it's for paying an order. Maybe in the future!

Reply:
Awesome. Thanks for the confirmation @juscamarena !

I went through the bitrefill shopping process today to check my thesis. It looks really good, so great job :+1:

Reply:
@jnewbery thanks for defending your thesis!

@juscamarena thank you very much for commenting!  I'll update the text.